### DISCUSSION TITLE
Tags/Metadata for foods

### DISCUSSION THREAD

**Discussion by Dilski** [mealie-recipes/mealie#4783] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

With the improved Home Assistant integration, I have been using Mealie in automations.

For example, for recipes scheduled for dinner which contains tofu I have a push notification reminding me to put the tofu in the tofu press in the afternoon. This is achieved by trying to match ingredients by ID in a script.

I'm also planning on trying the  "30 plants a week" challenge, and would like a way to tag ingredients as plants, so that I could have a script that scans the meals for the week and counts the different ingredients tagged as such.

### Please provide a concise description of the feature that would resolve your issue.

Similar to recipes, something like tags and/or API extra fields.

### Please consider and list out some caveats or tradeoffs made in your design decision

API extra fields may make more sense for my use case, as my goal is to do things with ingredients programmatically through the API.

Tags are more user-friendly and may get more use. Ingredient Tags could have visibility settings, (hide, show in recipe view, show in shopping list view, etc) so could be used to label allergens, brand preferences, etc.

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Dilski** [comment]
I've seen that the `CreateIngredientFood` and  `IngredientFood-Output` schemas in the api have extras in them, so this feature request is probably just a request for UI to set these values

**Comment by tmlmt** [comment]
In favor of this request 😃 

According to the [docs](https://docs.mealie.io/documentation/getting-started/api-usage/#shopping-list-and-food-extras):

> Similarly to recipes, extras are supported on shopping lists, shopping list items, and foods. At this time they are only accessible through the API.

So indeed the API and DB are ready but not the UI. How would it be best to implement that feature?

A separate "Foods" menu item on the left and therefore a dedicated page to edit foods?

**Comment by tmlmt** [comment]
This could be generalized to editing any field of foods. I just found out that there's actually a page where foods data can be edited, well hidden in the settings (/group/data/foods). But only few fields can be edited, and not tags / extras, ...
